### ISSUE TITLE
Unexpected exception during cancel operation.

### ISSUE THREAD

**Issue by singhsarab** [Microsoft/vstest#1661]
## Description
> This is a translation layer scenario. 
When cancellation request is raised by client, some times unexpected exceptions is reported back instead of clean cancel.
This is an inconsistent issue.

## Steps to reproduce
* Create a managed UT project in VS
* Add some tests that take some time to run
* Run All using Test Explorer
* Cancel the run via Test Explorer

## Expected behavior
> The cancel operation cancels the current run and no unexpected exceptions get logged.

## Actual behavior
> The following exception gets logged.

```
 Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: vstest.console process failed to connect to testhost process after 90 seconds. This may occur due to machine slowness, please set environment variable VSTEST_CONNECTION_TIMEOUT to increase timeout.

at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.ThrowExceptionOnConnectionFailure(Int32 connTimeout)

at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources)

at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)
```

**Comment by AbhitejJohn** [comment]
@singhsarab : Sure here you are: #1692

**Comment by singhsarab** [comment]
Thanks. Closing this one as it has been addressed via #1667
